src/Storage/Filters/RequestBodyTelemetryFilter.cs

```diff
@@ -0,0 +1,55 @@
+using System.Diagnostics.CodeAnalysis;
+using System.IO;
+using System.Net.Http;
+using System.Security.Claims;
+
+using AltinnCore.Authentication.Constants;
+
+using Microsoft.ApplicationInsights.Channel;
+using Microsoft.ApplicationInsights.DataContracts;
+using Microsoft.ApplicationInsights.Extensibility;
+using Microsoft.AspNetCore.Http;
+
+namespace Altinn.Platform.Storage.Filters
+{
+    /// <summary>
+    /// Filter to enrich request telemetry with request body for slow requests
+    /// </summary>
+    [ExcludeFromCodeCoverage]
+    public class RequestBodyTelemetryFilter : ITelemetryProcessor
+    {
+        private ITelemetryProcessor Next { get; set; }
+
+        private readonly IHttpContextAccessor _httpContextAccessor;
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="RequestBodyTelemetryFilter"/> class.
+        /// </summary>
+        public RequestBodyTelemetryFilter(ITelemetryProcessor next, IHttpContextAccessor httpContextAccessor)
+        {
+            Next = next;
+            _httpContextAccessor = httpContextAccessor;
+        }
+
+        /// <inheritdoc/>
+        public void Process(ITelemetry item)
+        {
+            RequestTelemetry request = item as RequestTelemetry;
+
+            if (request != null && request.Url.ToString().EndsWith("sbl/instances/search") && request.Duration.TotalMilliseconds > 1000)
+            {
+                var content = GetContentString(_httpContextAccessor.HttpContext.Request.Body);
+                request.Properties.Add("queryModel", content);
+            }
+
+            Next.Process(item);
+        }
+
+        private string GetContentString(Stream content)
+        {
+            var reader = new StreamReader(content);
+            reader.BaseStream.Seek(0, SeekOrigin.Begin);
```

Comment:
Would it be possible to create a test for this? Default behavior of the request stream is that it is none-reversible. We ran into issues with this before. I think there exists a property that enables seeking like this.

Reply:
Good point. Changed implementation to get the telemetry directly in the controller. This will result in logging the request body for all requests for a while. Not sure what is a higher cost logging all bodies or including a stopwatch to only retrieve the bodies of requests that run long. 
This approach results in the fewest number of modifications to the code